int32 index
---
int32 ret
